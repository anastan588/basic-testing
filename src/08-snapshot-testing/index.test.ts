// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const values = [5, 10, 15];
    const resultList = generateLinkedList(values);
    console.log(values);
    const neededLinkedList = {
      value: 5,
      next: {
        value: 10,
        next: {
          value: 15,
          next: {
            value: null,
            next: null,
          },
        },
      },
    };
    expect(resultList).toStrictEqual(neededLinkedList);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const values = [10, 15, 20];
    const resultList = generateLinkedList(values);
    expect(resultList).toMatchSnapshot();
  });
});
