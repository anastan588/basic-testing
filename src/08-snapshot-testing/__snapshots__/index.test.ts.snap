// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 1 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "c",
    },
    "value": "b",
  },
  "value": "a",
}
`;

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": 20,
    },
    "value": 15,
  },
  "value": 10,
}
`;
